internal/venice-common/src/main/java/com/linkedin/venice/authentication/AuthenticationService.java

```diff
@@ -0,0 +1,16 @@
+package com.linkedin.venice.authentication;
+
+import com.linkedin.venice.authorization.Principal;
+import java.security.cert.X509Certificate;
+import java.util.function.Function;
+
+
+/**
+ * Performs authentication.
+ */
+public interface AuthenticationService {
+  default Principal getPrincipalFromHttpRequest(X509Certificate certificate, Function<String, String> headers) {
```

Comment:
This patch is not adding lifecycle methods, like "init(configuration)" or "close()".
They will be added when we add the code that allows you to instantiate the AuthenticationService